hoomd/md/pytest/test_minimize_fire.py

```diff
@@ -0,0 +1,168 @@
+        'min_steps_conv': np.random.randint(1, 15)
+    }
+    return params
+
+
+def _set_and_check_new_params(fire):
+    """Set params to random values, then assert they are correct."""
+    # set the parameters to random values
+    new_params = _make_random_params()
+    for param in new_params:
+        setattr(fire, param, new_params[param])
+
+    # make sure they were set right
+    _assert_correct_params(fire, new_params)
+    return new_params
+
+
+def _assert_error_if_nonpositive(fire):
+    """Make sure error is raised if properties set to nonpositive values."""
+    negative_value = -np.random.randint(0, 26)
+    with pytest.raises(ValueError):
+        fire.min_steps_adapt = negative_value
+
+    with pytest.raises(ValueError):
+        fire.min_steps_conv = negative_value
+
+
+def test_constructor_validation():
+    """Make sure constructor validates arguments."""
+    with pytest.raises(ValueError):
+        md.minimize.FIRE(dt=0.01, aniso='nonsense')
+    with pytest.raises(ValueError):
+        md.minimize.FIRE(dt=0.01, min_steps_conv=-5)
+    with pytest.raises(ValueError):
+        md.minimize.FIRE(dt=0.01, min_steps_adapt=0)
+
+
+def test_get_set_params(simulation_factory, two_particle_snapshot_factory):
+    """Assert we can get/set params when not attached and when attached."""
+    fire = md.minimize.FIRE(dt=0.01)
+    default_params = {
+        'dt': 0.01,
+        'aniso': 'auto',
+        'min_steps_adapt': 5,
+        'finc_dt': 1.1,
+        'fdec_dt': 0.5,
+        'alpha_start': 0.1,
+        'fdec_alpha': 0.99,
+        'force_tol': 0.1,
+        'angmom_tol': 0.1,
+        'energy_tol': 1e-5,
+        'min_steps_conv': 10
+    }
+    _assert_correct_params(fire, default_params)
+
+    new_params = _set_and_check_new_params(fire)
+
+    _assert_error_if_nonpositive(fire)
+
+    # attach to simulation
+    snap = two_particle_snapshot_factory(d=2.34)
+    sim = simulation_factory(snap)
+    sim.operations.integrator = fire
+    sim.run(0)
+
+    # make sure the params are still right after attaching
+    _assert_correct_params(fire, new_params)
+
+    _set_and_check_new_params(fire)
+
+    _assert_error_if_nonpositive(fire)
+
+
+def test_run_minimization(lattice_snapshot_factory, simulation_factory):
+    """Run a short minimization simulation."""
+    snap = lattice_snapshot_factory(a=1.5, n=8)
+    sim = simulation_factory(snap)
+
+    lj = md.pair.LJ(default_r_cut=2.5, nlist=md.nlist.Cell())
+    lj.params[('A', 'A')] = dict(sigma=1.0, epsilon=1.0)
+    nve = md.methods.NVE(hoomd.filter.All())
+
+    fire = md.minimize.FIRE(dt=0.0025,
+                            methods=[nve],
+                            forces=[lj],
+                            min_steps_conv=3)
+
+    sim.operations.integrator = fire
+    sim.run(0)
+
+    initial_energy = fire.energy
+    steps_to_converge = 0
+    while not fire.converged:
+        sim.run(1)
+        steps_to_converge += 1
+
+    assert initial_energy >= fire.energy
```

Comment:
This test ensure that FIRE runs and does something. Could you also run an offline test and post screenshots of the before/after configurations of this run on the CPU and GPU so we can visually verify that it does appear to be minimizing the energy?